### ISSUE TITLE
Clarification needed: Use of network logits vs. predicted probabilities for calibration input in the framework

### ISSUE THREAD

**Issue by venk-na** [EFS-OpenSource/calibration-framework#63]
Hi,

I am currently working with the netcal calibration framework (EFS-OpenSource/calibration-framework) and studying the related paper "Multivariate Confidence Calibration for Object Detection".

The paper explicitly states that for calibration, one should use the network’s raw logits as input to the calibration mapping, aligning with classical logistic calibration formulations. From the paper in section 3.3 under "Independent Logistic Calibration:"
"However, instead of using the uncalibrated confidence estimate p^, we use the logit of the network as part of 
s to be conform with the original formulation in and ."

However, in the current calibration-framework detection-examples, it appears that the calibration methods operate directly on the predicted probabilities (from the method get_features()) rather than the raw logits.

I would like to receive some clarification:

Is there a recommended way to obtain or use the raw logits within the framework, or is it intended to work on predicted probabilities directly?

Could this difference affect calibration quality of the logistic calibration?

It would be very helpful if the documentation or examples could clarify this point.

Thank you very much for your work and support!

**Comment by fabiankueppers** [comment]
Hi @venk-na ,
please also take a look at https://github.com/EFS-OpenSource/calibration-framework/issues/28 where I already pointed out that the whole framework asserts probabilities/confidences and not logits. This has historical reasons to be aligned to common methods e.g. from scipy. Internally, these confidences are converted back to logits. Of course, recovering logits e.g. after a softmax does not recover the original logit completly as the formula is log(p) + c where c is an unknown constant. But tbh, I did not observe any major differences between calibration on logits and recovered logits. If you have any literature that supports or refutes this statement, please feel free to share it.

**Comment by venk-na** [comment]
Hi @fabiankueppers ,
Thank you for the quick response. Currently I am trying calibrate the yolov7 network. As you already know, the final confidence of the detection is obtained as a product as follows:

confidence = sigmoid(classification_logit) * sigmoid(objectness_logit)

Do you think using the current method where the provided confidences are converted to logits is still valid for the above use-case ?
If not, could you please suggest what could be done for such a class of detectors?
